core/src/main/java/tech/pegasys/web3signer/core/Runner.java

```diff
@@ -179,51 +168,82 @@ private void registerHttpHostAllowListHandler(final OpenAPI3RouterFactory openAp
 
   private OpenAPI3RouterFactory createOpenApiRouterFactory(
       final Vertx vertx,
-      final KeyIdentifiers keyIdentifiers,
+      final Set<String> blsIdentifiers,
+      final Set<String> secpIdentifiers,
       final MetricsSystem metricsSystem,
       final SignerForIdentifier<BlsArtifactSignature> blsSigner,
       final SignerForIdentifier<SecpArtifactSignature> secpSigner)
       throws InterruptedException, ExecutionException {
     final LogErrorHandler errorHandler = new LogErrorHandler();
     final OpenAPI3RouterFactory openAPI3RouterFactory = getOpenAPI3RouterFactory(vertx);
 
-    // upcheck handler
-    openAPI3RouterFactory.addHandlerByOperationId(
-        UPCHECK_OPERATION_ID, new BlockingHandlerDecorator(new UpcheckHandler(), false));
-    openAPI3RouterFactory.addFailureHandlerByOperationId(UPCHECK_OPERATION_ID, errorHandler);
-
-    // public key handler
-    openAPI3RouterFactory.addHandlerByOperationId(
-        GET_PUBLIC_KEYS_OPERATION_ID,
-        new BlockingHandlerDecorator(new GetPublicKeysHandler(keyIdentifiers), false));
-    openAPI3RouterFactory.addFailureHandlerByOperationId(
-        GET_PUBLIC_KEYS_OPERATION_ID, errorHandler);
+    // public key list handler
+    addEth1PublicKeysListHandler(openAPI3RouterFactory, secpIdentifiers, errorHandler);
```

Comment:
_personally_ prefer to think about this stuff as eth1 and eth2, rather than signing and public_keys APIs ... but that may be a personal preference in how to split the problem space.